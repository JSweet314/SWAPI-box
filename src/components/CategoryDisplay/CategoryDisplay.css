.category-display {
  display: flex;
  flex-wrap: wrap;
  height: 40rem;
  justify-content: center;
  width: 95%;
  background-color: #18212A;
  color: white;
  overflow: scroll;
}

.category-display--loading,
.category-display--default {
  align-items: center;
  justify-content: center;
}

.category-display--loading {
  background-color: #fff;
}

.category-display__default-text {
  margin: 0;
  font-size: 4rem;
  text-align: center;
}

.category-display h2 {
  width: 100%;
  text-align: center;
}