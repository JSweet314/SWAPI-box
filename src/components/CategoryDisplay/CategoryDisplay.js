import React from 'react';
import './CategoryDisplay.css';
import PropTypes from 'prop-types';
import loadingGIF from '../../images/Loading_icon.gif';
import InfoCard from '../InfoCard/index';

const CategoryDisplay = ({ categoryData, currentCategory, loading }) => {  
  if (loading) {
    return (
      <div className="category-display category-display--loading">
        <img src={loadingGIF} alt="loading" />
      </div>
    );
  }

  if (!categoryData.response) {
    return (
      <div className="category-display category-display--default">
        <p className="category-display__default-text">Select A Category</p>
      </div>
    );
  } 
  
  const response = categoryData.response.map(card => {
    return <InfoCard key={card.name} card={card} />;
  });

  return (
    <div className="category-display">
      <h2>{currentCategory}</h2>
      {response}
    </div>
  );
};

CategoryDisplay.propTypes = {
  categoryData: PropTypes.object.isRequired,
  currentCategory: PropTypes.string.isRequired,
  loading: PropTypes.bool.isRequired
};

export default CategoryDisplay;