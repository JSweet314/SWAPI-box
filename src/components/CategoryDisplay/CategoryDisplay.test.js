import React from 'react';
import { shallow } from 'enzyme';
import CategoryDisplay from './CategoryDisplay';

describe('CategoryDisplay', () => {
  it('should match a snapshot of the default text', () => {
    const wrapper = shallow(
      <CategoryDisplay 
        categoryData={[]}
        currentCategory={''}
        /* eslint-disable no-undef */
        changeFavCount={jest.fn()}
        selectFavorite={jest.fn()}
        removeFavorite={jest.fn()}
        /* eslint-enable no-undef */
        loading={true}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});