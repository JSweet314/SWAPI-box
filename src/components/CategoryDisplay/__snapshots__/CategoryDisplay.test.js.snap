// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryDisplay should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategoryDisplay />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="category-display__default-text"
>
        Select A Category
</p>,
      "className": "category-display",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Select A Category",
        "className": "category-display__default-text",
      },
      "ref": null,
      "rendered": "Select A Category",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="category-display__default-text"
>
          Select A Category
</p>,
        "className": "category-display",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select A Category",
          "className": "category-display__default-text",
        },
        "ref": null,
        "rendered": "Select A Category",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
